ci/get_hdf5_win.py

```diff
@@ -28,12 +30,22 @@
     "-DHDF5_BUILD_TOOLS:BOOL=OFF", "-DBUILD_TESTING:BOOL=OFF",
 ]
 if ZLIB_ROOT:
-    CMAKE_CONFIGURE_CMD += [
-        "-DHDF5_ENABLE_Z_LIB_SUPPORT=ON",
-        f"-DZLIB_INCLUDE_DIR={ZLIB_ROOT}\\include",
-        f"-DZLIB_LIBRARY_RELEASE={ZLIB_ROOT}\\lib_release\\zlib.lib",
-        f"-DZLIB_LIBRARY_DEBUG={ZLIB_ROOT}\\lib_debug\\zlibd.lib",
-    ]
+    if arch in ("arm64", "aarch64"):
+        # ZLIB includes based on vcpkg layout
+        CMAKE_CONFIGURE_CMD += [
+            "-DHDF5_ENABLE_Z_LIB_SUPPORT=ON",
+            f"-DZLIB_INCLUDE_DIR={ZLIB_ROOT}\\include",
+            f"-DZLIB_LIBRARY_RELEASE={ZLIB_ROOT}\\lib\\zlib.lib",
+            f"-DZLIB_LIBRARY_DEBUG={ZLIB_ROOT}\\debug\\lib\\zlibd.lib",
+        ]
+    else:
```

Comment:
should this be 
```suggestion
    elif arch in ("amd64", "x86_64"):
```

?
also, here too, please add a third, catch-all branch to raise a clear exception if we get an unexpected value, this will help finding the places that need an update in the future